opcodes/arc-opc.c

```diff
@@ -1695,7 +1695,7 @@ const struct arc_operand arc_operands[] =
 
   /* SIMM12_20 mask = 00000000000000000000111111222222.  */
 #define SIMM12_20	(UIMM6_20 + 1)
-  {12, 0, 0, ARC_OPERAND_SIGNED, insert_simm12_20, extract_simm12_20},
+  {12, 0, -SIMM12_20, ARC_OPERAND_SIGNED, insert_simm12_20, extract_simm12_20},
```

Comment:
There is a problem with this. When using -SIMM12_20 on an add2 that does not have @identifier@jli it throws an assertion because the internal relocation was not handled. As such, newlib won't even compile. Is there something extra I must do to handle this relocation before the assertion happens? I could instead check for -SIMM12_2 (which isn't defined in tc-arc.c) as the relocation type and replace it with BFD_RELOC_ARC_NONE (I would still need to define SIMM12_20 first - I used a hard coded number to test) :

```
diff --git a/gas/config/tc-arc.c b/gas/config/tc-arc.c
index c71bce4..c24e262 100644
--- a/gas/config/tc-arc.c
+++ b/gas/config/tc-arc.c
@@ -3879,6 +3879,11 @@ assemble_insn (const struct arc_opcode *opcode,
          if (insn->nfixups >= MAX_INSN_FIXUPS)
            as_fatal (_("too many fixups"));

+         if (reloc == -SIMM12_20)
+         {
+           reloc = BFD_RELOC_ARC_NONE;
+         }
+
          struct arc_fixup *fixup;
          fixup = &insn->fixups[insn->nfixups++];
          fixup->exp = *t;
```

Is there a better way?